### ISSUE TITLE
MessageSender abstraction is broken for Path when Send-Via is used

### ISSUE THREAD

**Issue by SeanFeldman** [Azure/azure-sdk-for-net#6031]
## Actual Behavior

`MessageSender` supports [Send-Via feature](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-transactions#transfers-and-send-via). Send-Via is using a transfer queue to route message to their final destination when using transactional processing. The abstraction is represented by a public overload of `MessageSender` that takes in two paths 
1. `entitytPath`
1. `viaEntityPath`
https://github.com/Azure/azure-service-bus-dotnet/blob/e278f87b0b3aa563ab055918e988baec9ec155a3/src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs#L127-L135

The abstraction has been copied from the old client and works, but it's broken when `Path` and `TransferDestinationPath` are queried back. That's because all public overloads are invoking an internal overload, swapping the two paths:
https://github.com/Azure/azure-service-bus-dotnet/blob/e278f87b0b3aa563ab055918e988baec9ec155a3/src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs#L132

where internal overload is
https://github.com/Azure/azure-service-bus-dotnet/blob/e278f87b0b3aa563ab055918e988baec9ec155a3/src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs#L137-L144

assigning `viaEntityPath` to `this.Path` and `entityPath` to `this.TransferDestinationPath`. Which is inverted.

## Expected Behavior
`Path` and Send-Via/`TransferDestinationPath` after `MessageSender` constructor is invoked should be identical to the values passed in.

## Repro

LinqPad [repro script](https://github.com/Azure/azure-service-bus-dotnet/files/2373545/asb.MessageSender.Path.vs.TransferDestinationPath.bug.linq.txt) attached.

```c#
var sender1 = new MessageSender(connection, "path");
sender1.Path.Dump("Should be 'path'");
Assert.IsNull(sender1.TransferDestinationPath); // green

var sender2 = new MessageSender(connection, "path", "intermediate");
Assert.AreEqual("path", sender2.Path); // red, equals to "intermediate"
Assert.AreEqual("intermediate", sender2.TransferDestinationPath); // red, equals to "path"
```

## Recommendations

The fix should be released as a **major version** for the following reasons:
1. This is a breaking change. While a fix will restore the correct behaviour, it will affect any existing solution relying on the incorrect implementation.
1. Anyone implementing a workaround will be broken if they update to a patch/minor.

**Comment by SeanFeldman** [comment]
@jsquire @AlexGhiondea have you had a chance to triage this one? This is a client bug.

**Comment by AlexGhiondea** [comment]
@SeanFeldman we will consider this for the next release of the Service Bus client. We currently do not have a release date for when that is going to be.

**Comment by SeanFeldman** [comment]
@AlexGhiondea note that this will be a breaking change and will require a new major.
In that case, you might want to introduce other breaking changes, such as https://github.com/Azure/azure-sdk-for-net/pull/6037.

**Comment by SeanFeldman** [comment]
@AlexGhiondea any updates about next (major) release and when this will be addressed?

**Comment by SeanFeldman** [comment]
@AlexGhiondea @jsquire would be nice to label this issue as `bug` to indicate what it is. Thank you.

**Comment by nemakam** [comment]
Closed in #6941

**Comment by SeanFeldman** [comment]
Needs to be a minor with the change made in PR.
https://github.com/Azure/azure-sdk-for-net/pull/6941#issuecomment-512110678
